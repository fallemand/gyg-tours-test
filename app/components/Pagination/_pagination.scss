@import '../../styles/variables';

.pagination {
  display: inline-block;

  .pagination__number {
    color: $button-color;
    font-weight: $font-light;
    font-size: 13px;
    padding: 8px 16px;
    text-decoration: none;
    transition: background-color .3s;
    border: 1px solid $button-background-color;
    border-radius: $border-radius;
    background-color: $button-background-color;
    margin: 0 4px;
    cursor: pointer;

    &.pagination__number--active {
      background-color: $button-background-color-active;
      border: 1px solid $button-background-color-active;

      &:focus {
        outline: none;
      }
    }

    &:hover:not(.pagination__number--active) {
      background-color: $button-background-color-active;
    }
  }
}
