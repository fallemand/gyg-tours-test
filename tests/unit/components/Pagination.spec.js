import React from 'react';
import Pagination from '../../../app/components/Pagination';

describe('Pagination', () => {
  const props = {
    total: 50,
    show: 5,
    onChange: jest.fn(),
    active: 1,
    className: '__CLASSNAME__',
  };

  it('render component', () => {
    const component = shallow(<Pagination {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('count amount of pages generated', () => {
    const component = shallow(<Pagination {...props} />);
    const buttons = component.find('.pagination__number');
    expect(buttons).toHaveLength((props.total / props.show) + 2);
  });

  it('click on another page should call onChange with page number', () => {
    const component = shallow(<Pagination {...props} />);
    // Active page is 1, so next number is 2.
    const pageTwoNumber = component.find('.pagination__number--active + .pagination__number');
    pageTwoNumber.simulate('click');
    expect(props.onChange).toHaveBeenCalledWith(2);
  });
});
