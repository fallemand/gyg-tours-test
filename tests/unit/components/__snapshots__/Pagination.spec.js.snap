// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination render component 1`] = `
"<div className='__CLASSNAME__ pagination'>
  <button type='button' className='pagination__number' onClick={[Function: onClick]}>
    «
  </button>
  <button type='button' className='pagination__number pagination__number--active' onClick={[Function: onClick]}>
    1
  </button>
  <button type='button' className='pagination__number' onClick={[Function: onClick]}>
    2
  </button>
  <button type='button' className='pagination__number' onClick={[Function: onClick]}>
    3
  </button>
  <button type='button' className='pagination__number' onClick={[Function: onClick]}>
    4
  </button>
  <button type='button' className='pagination__number' onClick={[Function: onClick]}>
    5
  </button>
  <button type='button' className='pagination__number' onClick={[Function: onClick]}>
    6
  </button>
  <button type='button' className='pagination__number' onClick={[Function: onClick]}>
    7
  </button>
  <button type='button' className='pagination__number' onClick={[Function: onClick]}>
    8
  </button>
  <button type='button' className='pagination__number' onClick={[Function: onClick]}>
    9
  </button>
  <button type='button' className='pagination__number' onClick={[Function: onClick]}>
    10
  </button>
  <button type='button' className='pagination__number' onClick={[Function: onClick]}>
    »
  </button>
</div>"
`;
